packages/runtime/client/rest/src/contract/endpoints/sources.ts

```diff
@@ -40,3 +40,94 @@ export const createContractSources = (axiosInstance: AxiosInstance) => {
     );
   };
 };
+
+export interface GetContractBySourceIdRequest {
+  contractSourceId: BuiltinByteString;
+  expand?: boolean;
+}
+
+export type GetContractBySourceIdResponse = Contract;
+
+const GetContractBySourceIdResponseGuard: t.Type<GetContractBySourceIdResponse> =
+  G.Contract;
+
+export const getContractSourceById =
+  (axiosInstance: AxiosInstance) =>
+  async ({
+    contractSourceId,
+    expand,
+  }: GetContractBySourceIdRequest): Promise<GetContractBySourceIdResponse> => {
+    const response = await axiosInstance.get(
+      `/contracts/sources/${encodeURIComponent(contractSourceId)}`,
+      { params: { expand } }
+    );
+    return pipe(
+      GetContractBySourceIdResponseGuard.decode(response.data),
+      E.match(
+        (e) => {
+          throw formatValidationErrors(e);
+        },
+        (e) => e
+      )
+    );
+  };
+
+export interface GetContractSourceAdjacencyRequest {
+  contractSourceId: BuiltinByteString;
```

Comment:
I believe a type alias might clarify the intent.
```
type ContractSourceId = string 
// or 
type ContractSourceId = BuiltinByteString
```

We could asses also if a Branded type makes sense to show that not all strings are the same

Reply:
Have we figured out yet how we want to do branded types?

Reply:
Yes, using t.brand from io-ts